File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/utils/MdcUtility.java
package uk.gov.justice.digital.hmpps.keyworker.utils;

import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class MdcUtility {

    public static final String USER_ID_HEADER = "userId";
    public static final String REQUEST_ID = "requestId";
    public static final String REQUEST_DURATION = "duration";
    public static final String RESPONSE_STATUS = "status";

    public String generateUUID() {
        return UUID.randomUUID().toString();
    }

}

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/utils/MdcUtility.kt
package uk.gov.justice.digital.hmpps.keyworker.utils

import org.slf4j.MDC

object MdcUtility {
  const val USER_ID_HEADER = "userId"
  const val REQUEST_DURATION = "duration"
  const val RESPONSE_STATUS = "status"
  const val SKIP_LOGGING = "skipLogging"
  val isLoggingAllowed: Boolean
    get() = "true" != MDC.get(SKIP_LOGGING)
}
